api/cache/object_cache.go

```diff
@@ -0,0 +1,62 @@
+package cache
+
+import (
+	"time"
+
+	"github.com/bluele/gcache"
+	"github.com/nspcc-dev/neofs-api-go/pkg/object"
+)
+
+// ObjectsCache provides interface for lru cache for objects.
+type ObjectsCache interface {
+	Get(address *object.Address) *object.Object
+	Put(address *object.Address, obj object.Object) error
```

Comment:
`object.Object` should already have CID/OID inside. Why do we need to have the same in `address`?

Reply:
Because when we put an object into the cache we already have parsed object address via:
```
addr := object.NewAddress()
	addr.SetObjectID(oid)
	addr.SetContainerID(bkt.CID)
```

I'm not sure we should do the same inside Put function again and don't use the ready-to-use value.